crates/biome_service/src/workspace/server.rs

```diff
@@ -75,7 +75,7 @@ pub(crate) struct Document {
     pub(crate) file_source_index: usize,
 
     /// The result of the parser (syntax tree + diagnostics).
-    pub(crate) syntax: AnyParse,
+    pub(crate) syntax: Result<AnyParse, FileTooLarge>,
```

Comment:
Maybe the document can have a new field called `diagnostics` instead. This could be useful to collect more than one diagnostic related to this document (file size, ignored, unknown document, etc.

Reply:
Unlike parser errors, of which there can be multiple in a document, I think there will be only one reason that would prevent parsing altogether. That’s why I went for a `Result` type, since it clearly communicates we bailed out before we even got to the parsing.

In theory we could use a `Vec` of diagnostics in the `Err` variant, and if we ever need to it’s an easy change. I’m just not convinced yet we _will_ need to.

Reply:
On second consideration, maybe an [`Either`](https://docs.rs/either/latest/either/) could communicate the semantics even slightly clearer than a `Result`. After all, it wasn't an error of the parser; instead, we **either** parse or store a diagnostic. But I wonder if this subtle semantic difference is worth the extra dependency :sweat_smile: